pkg/query/flamegraph_arrow.go

```diff
@@ -199,13 +198,13 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 
 	cumulative := int64(0)
 	height := int32(0)
-	rootsRow := []uint32{}
-	children := make([][]uint32, len(p.Samples))
+	rootsRow := []int{}
+	children := make(map[int][]int, len(p.Samples))
```

Comment:
This we should undo I think, but if you rather wanted to commit what works and improve I totally understand